firewall/inspection/portscan/detect.go

```diff
@@ -0,0 +1,357 @@
+			ipClass == netutils.SiteLocal ||
+			ipClass == netutils.Invalid ||
+			ipClass == netutils.Global) &&
+		!isNetBIOSoverTCPIP(conn) &&
+		!(conn.IPProtocol == packet.UDP && (conn.LocalPort == 67 || conn.LocalPort == 68)) { // DHCP
+
+		handleMaliciousPacket(ctx, inMap, conn, entry, ipString)
+	}
+
+	if inMap && entry.blocked {
+		log.Tracer(ctx).Debugf("blocking")
+		conn.SetVerdict(network.VerdictDrop, "Portscan", nil)
+	} else {
+		log.Tracer(ctx).Debugf("let through")
+	}
+
+	return network.VerdictUndecided, false, nil
+}
+
+func handleMaliciousPacket(ctx context.Context, inMap bool, conn *network.Connection, entry *ipData, ipString string) {
+	//define Portscore
+	var addScore int
+	switch {
+	case conn.LocalPort < startRegisteredPorts:
+		addScore = addScoreWellKnownPort
+	case conn.LocalPort < startDynamicPorts:
+		addScore = addScoreRegisteredPort
+	default:
+		addScore = addScoreDynamicPort
+	}
+
+	if !inMap {
+		//new IP => add to List
+		ips[ipString] = &ipData{
+			score: addScore,
+			blockedPorts: []tcpUDPport{
+				tcpUDPport{
+					protocol: conn.IPProtocol,
+					port:     conn.LocalPort,
+				},
+			},
+			lastSeen:    time.Now(),
+			lastUpdated: time.Now(),
+		}
+		log.Tracer(ctx).Debugf("New Entry: %+v", ips[ipString])
+	} else {
+		//Port in list of tried ports?
+		triedPort := false
+		port := tcpUDPport{protocol: conn.IPProtocol, port: conn.LocalPort}
+		for _, e := range entry.blockedPorts {
+			if e == port {
+				triedPort = true
+				break
+			}
+		}
+
+		if !triedPort {
+			entry.blockedPorts = append(entry.blockedPorts, tcpUDPport{protocol: conn.IPProtocol, port: conn.LocalPort})
+			entry.score = intMin(entry.score+addScore, maxScore)
+
+			if entry.previousOffender || entry.score >= scoreBlock {
+				entry.blocked = true
+				entry.previousOffender = true
+
+				//TODO: actually I just want to know if THIS threat exists - I don't need prefixing. Maybe we can do it simpler ... (less CPU-intensive)
+				if t, _ := status.GetThreats(threadPrefix + ipString); len(t) == 0 {
+					log.Tracer(ctx).Debugf("new Threat")
+					status.AddOrUpdateThreat(&status.Threat{
+						ID:              threadPrefix + ipString,
+						Name:            "Portscan by " + ipString,
+						Description:     "Someone tries to connect to a lot of closed Ports (non-running Services). Probably he wants to find out the services running on the maschine to determine which services to attack", //fixme: to long
+						MitigationLevel: status.SecurityLevelHigh,
+						Started:         time.Now().Unix(),
+					})
+				}
+			}
+		}
+
+		log.Tracer(ctx).Debugf("changed Entry: %+v", entry)
+	}
+}
+
+//updateScoreIgnoreBlockPrevOffender updates this 4 Values of the Struct
+//ipString needs to correspond to the key of the entry in the map ips
+//WARNING: This function maybe deletes the entry ipString from the Map ips. (look at the returncode)
+//return: still in map? (bool)
+func (d *ipData) updateScoreIgnoreBlockPrevOffender(ipString string) bool {
```

Comment:
Please add a leading space to comment strings, it's much easier to read this way. Also, I understand why you chose the name of the function but it's not really obvious what's the actual purpose (rather than updating some obscure values).